src/topology/algebra/uniform_convergence.lean

```diff
@@ -18,29 +19,39 @@ space of continuous linear maps between two topological vector spaces.
 * `uniform_convergence.uniform_group` : if `G` is a uniform group, then the uniform structure of
   uniform convergence makes `α → G` a uniform group
 * `uniform_convergence_on.uniform_group` : if `G` is a uniform group, then the uniform structure of
-  `𝔖`-convergence, for any `𝔖 : set (set α)`, makes `α → G` a uniform group
+  `𝔖`-convergence, for any `𝔖 : set (set α)`, makes `α → G` a uniform group.
+* `uniform_convergence_on.has_continuous_smul_of_image_bounded` : let `E` be a TVS,
+  `𝔖 : set (set α)` and `H` a submodule of `α → E`. If the image of any `S ∈ 𝔖` by any `u ∈ H` is
+  bounded (in the sense of `bornology.is_vonN_bounded`), then `H`, equipped with the topology of
+  `𝔖`-convergence, is a TVS.
 
 ## TODO
 
-* Let `E` be a TVS, `𝔖 : set (set α)` and `H` a submodule of `α → E`. If the image of any `S ∈ 𝔖`
-  by any `u ∈ H` is bounded (in the sense of `bornology.is_vonN_bounded`), then `H`, equipped with
-  the topology of `𝔖`-convergence, is a TVS.
+* `uniform_convergence_on.has_continuous_smul_of_image_bounded` unnecessarily asks for `𝔖` to be
+  nonempty and directed. This will be easy to solve once we know that replacing `𝔖` by its
+  ***noncovering*** bornology (i.e ***not*** what `bornology` currently refers to in mathlib)
+  doesn't change the topology.
 
 ## References
 
 * [N. Bourbaki, *General Topology, Chapter X*][bourbaki1966]
+* [N. Bourbaki, *Topological Vector Spaces*][bourbaki1987]
 
 ## Tags
 
 uniform convergence, strong dual
 
 -/
 
+open filter
+open_locale topological_space pointwise
+
 section group
 
-variables {α G : Type*} [group G] [uniform_space G] [uniform_group G] {𝔖 : set $ set α}
+variables {α G ι : Type*} [group G] [uniform_space G] [uniform_group G] {𝔖 : set $ set α}
 
 local attribute [-instance] Pi.uniform_space
+local attribute [-instance] Pi.topological_space
 
```

Comment:
You can save yourself a little headache and improve readability with this (and end the section before `uniform_convergence_on`). This can avoid all the `@nhds` and `letI`'s and things prior to `uniform_convergence_on`.
```suggestion

section uniform_convergence

local attribute [instance] uniform_convergence.uniform_space

```

Reply:
This is actually what I was doing at first, but Sébastien [suggested](https://github.com/leanprover-community/mathlib/pull/14693#discussion_r970558294) avoiding it when reviewing #14693 because this could be ambiguous to readers, and I think that as long as we don't have a better way to clearly express we're talking about uniform convergence it's better to keep things explicit.